src/tokens/tokens.service.ts

```diff
@@ -338,65 +343,79 @@ export class TokensService {
     return basicAuth(this.username, this.password);
   }
 
-  private async queryPool(poolLocator: IValidPoolLocator) {
-    const schema = poolLocator.schema as ContractSchemaStrings;
+  private async wrapError<T>(response: Promise<AxiosResponse<T>>) {
+    return response.catch(err => {
+      if (axios.isAxiosError(err)) {
+        const errorMessage = err.response?.data?.error;
+        throw new InternalServerErrorException(errorMessage ?? err.message);
```

Comment:
I note we're translating all status codes to 500 here, regardless of the backend error code.
Given that - should we log details the error in this layer before we throw it?

Reply:
... or is there already an interceptor installed to axios that logs the HTTP calls in and out for that?

Reply:
There isn't. I also didn't know axios interceptors were a thing. Perhaps I should be using those instead of this `wrapError` method.

Reply:
See the logger in here - think that's great for now